swift/Sources/Keystore/KeyStore.swift

```diff
@@ -0,0 +1,285 @@
+        self.keyDirectory = keyDirectory
+        try load()
+    }
+
+    private func load() throws {
+        let fileManager = FileManager.default
+        try? fileManager.createDirectory(at: keyDirectory, withIntermediateDirectories: true, attributes: nil)
+
+        let accountURLs = try fileManager.contentsOfDirectory(at: keyDirectory, includingPropertiesForKeys: [], options: [.skipsHiddenFiles])
+        for url in accountURLs {
+            guard let key = StoredKey.load(path: url.path) else {
+                // Ignore invalid keys
+                continue
+            }
+            let wallet = Wallet(keyURL: url, key: key)
+            wallets.append(wallet)
+        }
+    }
+
+    /// Creates a new wallet. HD default by default
+    public func createWallet(password: String, coins: [CoinType]) throws -> Wallet {
+        let key = StoredKey(password: password)
+        return try saveCreatedWallet(for: key, password: password, coins: coins)
+    }
+
+    private func saveCreatedWallet(for key: StoredKey, password: String, coins: [CoinType]) throws -> Wallet {
+        let url = makeAccountURL()
+        let wallet = Wallet(keyURL: url, key: key)
+        for coin in coins {
+            _ = try wallet.getAccount(password: password, coin: coin)
+        }
+        wallets.append(wallet)
+
+        try save(wallet: wallet)
+
+        return wallet
+    }
+
+    /// Adds accounts to a wallet.
+    public func addAccounts(wallet: Wallet, coins: [CoinType], password: String) throws -> [Account] {
+        var accounts = [Account]()
+        for coin in coins {
+            accounts.append(try wallet.getAccount(password: password, coin: coin))
+        }
+        try save(wallet: wallet)
+        return accounts
+    }
+
+    /// Imports an encrypted JSON key.
+    ///
+    /// - Parameters:
+    /// - json: json wallet
+    /// - password: key password
+    /// - newPassword: password to use for the imported key
+    /// - coin: coin to use for this wallet
+    /// - Returns: new account
+    public func `import`(json: Data, password: String, newPassword: String, coin: CoinType) throws -> Wallet {
+        guard let key = StoredKey.importJSON(json: json) else {
+            throw Error.invalidKey
+        }
+        guard let data = key.decryptPrivateKey(password: password) else {
+            throw Error.invalidPassword
+        }
+
+        if let mnemonic = checkMnemonic(data) {
+            return try self.import(mnemonic: mnemonic, encryptPassword: newPassword, coin: coin)
+        }
+
+        guard let privateKey = PrivateKey(data: data) else {
+            throw Error.invalidKey
+        }
+        return try self.import(privateKey: privateKey, password: newPassword, coin: coin)
+    }
+
+    private func checkMnemonic(_ data: Data) -> String? {
+        guard let mnemonic = String(data: data, encoding: .ascii) else {
+            return nil
+        }
+
+       if !HDWallet.isValid(mnemonic: mnemonic) {
```

Comment:
I would add this check in guard statement

Reply:
done